frontend/src/redux/reducers/ui.tsx

```diff
@@ -188,6 +192,24 @@ function reducer(state = _.cloneDeep(INITIAL_STATE), action: Action) {
       newState.sidebar = { ...newState.sidebar, ...setInitialSidebarOpen() };
       return newState;
     }
+    case UI_SET_NOTIFICATIONS: {
+      const notifications = action.notifications;
+      /* There are two ways user can send notifications either a complete set of array or a single notification
+         when handling the array notifications we want to only have unique set of notifications pushed into the UI notifications config
+      */
+      if (Array.isArray(notifications)) {
+        const uniqueNotifications = _.uniqBy(
+          [...notifications].concat(newFilters.notifications),
+          function (item) {
+            return item.id;
```

Comment:
You could remove the extra dependency and all the id stuff by using:
```ts
return `${item.date},${item.message}`
```

There's no need for a UUID, since no UUID properties are being used.